bintray/services/upload.go

```diff
@@ -96,6 +99,15 @@ func (us *UploadService) uploadFiles(uploadDetails *UploadParams, artifacts []cl
 					if uploaded {
 						uploadCount[threadId]++
 					}
+					if uploadDetails.ListDownloads {
+						// This needs to not sleep, but since the publish is async we need to for now
+						time.Sleep(15 * time.Second)
```

Comment:
I understand the reason for putting this 15 seconds sleep, but I'm not sure this solution is reliable enough. We're not sure whether 15 seconds is enough or maybe it is too much and causes the operations to be delayed when it could have finished sooner.
Here's a suggested alternative for this:

First, we can send the "add to download list" requests after all files are uploaded, in the main routine, instead of after every file upload as part fo the parallel upload routines. This can increase the chances that each individual file is already published by the time we try to publish it.

The above still does not guarantee that the file is published by the time we try to list it. We can add a retry mechanism as follows:
1. Try to list all published file in a loop (with no sleep).
2. Only if there some files which are unpublished, sleep for a second, and try again #1 (for the remaining files only).